File: content/04/code/dotnet/04--step04-empty-scenarios-added/Shouty/Shouty/Person.cs
using System;
using System.Collections.Generic;

namespace Shouty
{
    public class Person
    {
        private readonly Network network;
        private readonly List<string> messagesHeard = new List<string>();

        public Person(Network network)
        {
            this.network = network;
            network.Subscribe(this);
        }

        public void MoveTo(int distance)
        {
        }

        public void Shout(string message)
        {
            network.Broadcast(message);
        }

        public IList<string> GetMessagesHeard()
        {
            return messagesHeard;
        }

        public void Hear(string message)
        {
            messagesHeard.Add(message);
        }
    }
}

File: content/bdd-with-cucumber/09/code/java/03-add-dependency-on-picocontainer/shouty/src/main/java/shouty/Person.java
package shouty;

import java.util.ArrayList;
import java.util.List;

public class Person {
    private final List<String> messagesHeard = new ArrayList<String>();
    private final Network network;
    private final String name;
    private int location;
    private int credits;

    public Person(String name, Network network, int location) {
        this.name = name;
        this.network = network;
        this.location = location;
        this.credits = 0;
        network.subscribe(this);
    }

    public List<String> getMessagesHeard() {
        return messagesHeard;
    }

    public void shout(String message) {
        network.broadcast(message, this);
    }

    public void hear(String message) {
        messagesHeard.add(message);
    }

    public int getLocation() {
        return location;
    }

    public void setCredits(int credits) {
        this.credits = credits;
    }

    public Person moveTo(int newLocation) {
        this.location = newLocation;
        return this;
    }

    public int getCredits() {
        return credits;
    }

    public String getName() { return name; }
}
